### DISCUSSION TITLE
Remove publish option for schools

### DISCUSSION THREAD

**Discussion by sjquigley** [phcode-dev/phoenix#1606] [General]
Schools in the US are very concerned about privacy issues and bullying. It seems like the 30-day "publish" function could create some problems and even libel lawsuits. Thoughts on this? Are there any options for removing this function?

**Comment by abose** [comment]
Hello @sjquigley Thanks for checking out Phoenix Code and reaching out with valuable feedback.

I received your email too reg. this and replied there too.


The publish feature has its ups and downs. Since we don't have any logins, a restriction was placed to initially delete sites after 7 days. But then, it was a school which requested to increase the time to 30 days so that teachers get time to evaluate and grade student projects. There was also questions relating to organizational login for schools, students and teachers to manage group learning and shared work spaces .We have been asked about building "Phoenix For Education" with said features, which we have not yet prioritized.


Reg. the publish workflow, Phoenix Code Desktop builds are only about 1.5 months old(web builds have been around for a year) and till now, the focus was on creating the app. We are working on integrating logins to gate publishing by June, post which you would need an account to publish content(but you can still use the app without any accounts). 


Will the above login requirement fix your use case, or is there something more that you are looking for? Also, any feedback on education related requirements/workflows will help us prioritize that.

**Comment by sjquigley** [comment]
Thanks Arun et al.

We love Phoenix Code and are really excited about promoting it amongst secondary ELA teachers in the US.

*Yes, I think creating a login function for publishing is super useful and will help flag/remove problematic content. And thank you for making this FOSS

*Most US schools use Chromebook computers. I was wondering about the possibility of developing a Chromebook app that does not require a linux development environment. School systems are very restrictive about what apps they allow on student computers and I doubt they allow Linux development environments.

Ideally, we would want an IT school specialist managing 2,000 Chromebooks to be able to add Phoenix Code App to all district computers in one click.

On another note, I have noticed that Phoenix Code loads in the browser via cached files….even without internet. We had set up all of our computers with a Phoenix Code icon shortcut located in app launcher. However, this shortcut function was removed in the latest Chrome OS release. :(.  As a workaround for the latest Chrome OS release, we “pinned” Phoenix Code to all of our browsers. All of this is a hassle. It would be great to have a Phoenix Code Icon in the App Launcher, even if it’s just a browser shortcut.







Kind regards,

Stephen J. Quigley, Ph.D.
John C. Mascaro Faculty Lecturer in Sustainability
Associate Teaching Professor
University of Pittsburgh
English Department
Cathedral of Learning 628K
http://www.pitt.edu/~sjq4
***@***.******@***.***>

Pitt Senate Computing and
Information Technology
Committee Member

Have you heard about Pitt’s
new Digital Narrative and
Interactive Design Major?
https://www.dnid.pitt.edu































On May 7, 2024, at 9:41 PM, Arun Bose ***@***.***> wrote:



Hello @sjquigley<https://github.com/sjquigley> Thanks for checking out Phoenix Code and reaching out with valuable feedback.

I received your email too reg. this and replied there too.

The publish feature has its ups and downs. Since we don't have any logins, a restriction was placed to initially delete sites after 7 days. But then, it was a school which requested to increase the time to 30 days so that teachers get time to evaluate and grade student projects. There was also questions relating to organizational login for schools, students and teachers to manage group learning and shared work spaces .We have been asked about building "Phoenix For Education" with said features, which we have not yet prioritized.

Reg. the publish workflow, Phoenix Code Desktop builds are only about 1.5 months old(web builds have been around for a year) and till now, the focus was on creating the app. We are working on integrating logins to gate publishing by June, post which you would need an account to publish content(but you can still use the app without any accounts).

Will the above login requirement fix your use case, or is there something more that you are looking for? Also, any feedback on education related requirements/workflows will help us prioritize that.

**Reply by abose** [reply]
Thanks for the detailed Notes. Discussed this with the team and we moved this issue to high priority for June release.

> Most US schools use Chromebook computers. I was wondering about the possibility of developing a Chromebook app that does not require a Linux development environment. 

1. Will ship Phoenix Code as a Chrome OS app and full first party support of Chrome OS by June. Will share the builds in this discussion when available for feedback. 

2. Publish feature will be gated via logins by June-2024

> On another note, I have noticed that Phoenix Code loads in the browser via cached files….even without internet. 

3.  That is correct, for faster app loads, we cache the assets locally. But i guess a native Chrome OS app would be better suited instead of playing catch-up with OS Updates in a managed deployment environment like in Schools.

Are there any timelines we should be aware of? Also are there any other features we should include apart from the above two?

>  And thank you for making this FOSS
   Any time! :smiley:

**Reply by abose** [reply]
Hi @sjquigley ,
 We investigated into adding native chrome app support and it is doable, but this will be a major open source development investment from our end to support chromeOS, and its continued operational support.

 We are bought into your proposal, and would also like to add chromeOS support for education as it aligns with our goal of making coding accessible and playful to new coders.

 Before we go ahead with the development, i would like to get your confirmation that this is something that you will be interested in and will be using. Our open source development resources are limited and it is essential for us to prioritize where we spend our dev efforts. 

Please also provide any timelines that you are looking at, so that we can prioritize accordingly.

Best,
Arun

**Comment by sjquigley** [comment]
Hello Arun,

Yes, I think working on a Chrome App makes sense. Especially for school districts with bad wifi.

As far as our research and development, yes, Phoenix Code is our recommended code editor.

Just finished the video tutorial:


<https://youtu.be/j8o506PZ1H8>
[hqdefault.jpg]
Codela Phoenix Code Editor Tutorial<https://youtu.be/j8o506PZ1H8>
youtu.be<https://youtu.be/j8o506PZ1H8>





Kind regards,

Stephen J. Quigley, Ph.D.
John C. Mascaro Faculty Lecturer in Sustainability
Associate Teaching Professor
University of Pittsburgh
English Department
Cathedral of Learning 628K
http://www.pitt.edu/~sjq4
***@***.******@***.***>

Pitt Senate Computing and
Information Technology
Committee Member

Have you heard about Pitt’s
new Digital Narrative and
Interactive Design Major?
https://www.dnid.pitt.edu































On Jun 3, 2024, at 2:30 AM, Arun Bose ***@***.***> wrote:



Hi @sjquigley<https://github.com/sjquigley> ,
We investigated into adding native chrome app support and it is doable, but this will be a major open source development investment from our end to support chromeOS, and its continued operational support.

We are bought into your proposal, and would also like to add chromeOS support for education as it aligns with our goal of making coding accessible and playful to new coders.

Before we go ahead with the development, i would like to get your confirmation that this is something that you will be interested in and will be using. Our open source development resources are limited and it is essential for us to prioritize where we spend our dev efforts.

Please also provide any timelines that you are looking at, so that we can prioritize accordingly.

Best,
Arun

**Reply by abose** [reply]
Thanks for confirming. 

Looks the chrome app is now going to be one of the supported platforms @charlypa

**Comment by charlypa** [comment]
@sjquigley
We've built a Chrome app for Phoenix Code and need to meet Google Play Store's requirement of 21 beta testers for publication. Finding local testers has been challenging due to the high usage of Linux in our area. Could you help us by providing a list of 21 beta testers willing to provide feedback? This would enable us to proceed with publishing the app.

**Reply by abose** [reply]
@sjquigley we can move this conversion to email as we need email address to forward the chrone os app install invite. I have sent an email to your mail address for more details

**Comment by sjquigley** [comment]
Hello All,

1. We are obviously looking for Chrome OS users. Is this correct?

2. What prior knowledge does this user need?

3. What tasks will this user need to perform? Do you have a sample testing script?

4. How will you gather testing data? Screen recordings, etc?

Most adults do not use Chrome OS, but most of our schools do use Chrome OS. These computers are often locked down by school systems. My lab has several Chrome computers I can loan to people for testing. I just need to know what you would need on my end, or how else I can assist in the process.




Kind regards,

Stephen J. Quigley, Ph.D.
John C. Mascaro Faculty Lecturer in Sustainability
Associate Teaching Professor
University of Pittsburgh
English Department
Cathedral of Learning 628K
http://www.pitt.edu/~sjq4
***@***.******@***.***>

Pitt Senate Computing and
Information Technology
Committee Member

Have you heard about Pitt’s
new Digital Narrative and
Interactive Design Major?
https://www.dnid.pitt.edu































On Jun 17, 2024, at 1:32 PM, charly p abraham ***@***.***> wrote:



@sjquigley<https://github.com/sjquigley>
We've built a Chrome app for Phoenix Code and need to meet Google Play Store's requirement of 21 beta testers for publication. Finding local testers has been challenging due to the high usage of Linux in our area. Could you help us by providing a list of 21 beta testers willing to provide feedback? This would enable us to proceed with publishing the app.

**Reply by abose** [reply]
Hi @sjquigley we are doing some early testing at our end, will get back to you next week with the steps. The high level tasks are:

1. We need a list of 21 email addresses who we need to send the app install invite to. The chrome app can be installed from those email addresses from the play store.
2. A short review needs to be submitted to the play store.

@charlypa will follow up with the detailed steps next week. Thanks for helping with the tests.

**Comment by charlypa** [comment]
Hello @sjquigley,

> 1. We are obviously looking for Chrome OS users. Is this correct?

Yes, we are currently focusing on making the app exclusively available for Chrome OS users through the Play Store.

> 2. What prior knowledge does this user need?

Users should be comfortable with downloading and installing apps from the Play Store. They will need to follow an email invitation to install the app and then provide feedback through a specified link.

> 3. What tasks will this user need to perform? Do you have a sample testing script?

The tasks will involve downloading the app from the Play Store upon receiving an email invite, installing it, and submitting feedback via a specified link. Currently, we do not have a testing script available.

> 4. How will you gather testing data? Screen recordings, etc?

We plan to collect feedback through user reviews and ratings submitted in English on the Play Store. This information will be utilized by both our team and Google to assess the app's performance and user experience.

We need the emails of 21 volunteers willing to try the app and provide feedback. Once we have these emails, we can send out invites. After receiving their feedback, we will request Google to review and publish it on the Play Store.

**Comment by sjquigley** [comment]
I understand. Can the user search for the Phoenix App on the Play Store, or is an email required?



Kind regards,

Stephen J. Quigley, Ph.D.
John C. Mascaro Faculty Lecturer in Sustainability
Associate Teaching Professor
University of Pittsburgh
English Department
Cathedral of Learning 628K
http://www.pitt.edu/~sjq4
***@***.******@***.***>

Pitt Senate Computing and
Information Technology
Committee Member

Have you heard about Pitt’s
new Digital Narrative and
Interactive Design Major?
https://www.dnid.pitt.edu































On Jun 23, 2024, at 11:48 AM, charly p abraham ***@***.***> wrote:



Hello @sjquigley<https://github.com/sjquigley>,

  1.  We are obviously looking for Chrome OS users. Is this correct?

Yes, we are currently focusing on making the app exclusively available for Chrome OS users through the Play Store.

  1.  What prior knowledge does this user need?

Users should be comfortable with downloading and installing apps from the Play Store. They will need to follow an email invitation to install the app and then provide feedback through a specified link.

  1.  What tasks will this user need to perform? Do you have a sample testing script?

The tasks will involve downloading the app from the Play Store upon receiving an email invite, installing it, and submitting feedback via a specified link. Currently, we do not have a testing script available.

  1.  How will you gather testing data? Screen recordings, etc?

We plan to collect feedback through user reviews and ratings submitted in English on the Play Store. This information will be utilized by both our team and Google to assess the app's performance and user experience.

We need the emails of 21 volunteers willing to try the app and provide feedback. Once we have these emails, we can send out invites. After receiving their feedback, we will request Google to review and publish it on the Play Store.

**Reply by charlypa** [reply]
@sjquigley

Yes, an emails is required. The app won't be visible until it is published in the Play Store. This is a prerequisite for submitting the app for review to Google.

**Comment by sjquigley** [comment]


**Reply by abose** [reply]
Hi @sjquigley replying to the email will post the email in github exposing the emails in github comments. Github usually masks this email for privacy and safety, so we are unable to see any of the email addresses. I will send over an email privately. Please reply to that email with the mail addresses.

**Comment by charlypa** [comment]
@sjquigley 
We have submitted the app for review to Google for open testing. Once its approved, I will share it across

**Comment by charlypa** [comment]
@sjquigley 
Phoenix Code is now available in google play store for Chromebook

**Comment by sjquigley** [comment]
Well done! This chrome app is an important accessibility feature.



Kind regards,

Stephen J. Quigley, Ph.D.
John C. Mascaro Faculty Lecturer in Sustainability
Associate Teaching Professor
University of Pittsburgh
English Department
Cathedral of Learning 628K
http://www.pitt.edu/~sjq4
***@***.******@***.***>

Pitt Senate Computing and
Information Technology
Committee Member

Have you heard about Pitt’s
new Digital Narrative and
Interactive Design Major?
https://www.dnid.pitt.edu































On Jul 30, 2024, at 1:28 AM, charly p abraham ***@***.***> wrote:



@sjquigley<https://github.com/sjquigley>
Phoenix Code is now available in google play store for Chromebook

**Reply by charlypa** [reply]
@sjquigley 
Please provide feedback  if you face any issues

**Comment by charlypa** [comment]
Hello @sjquigley,

You can find the code used for the app at this repository: [phoenix-android on GitHub](https://github.com/phcode-dev/phoenix-andriod.git).

Feel free to check it out and let me know if you have any questions or need further assistance!

**Reply by charlypa** [reply]
Hello @sjquigley 

Have you got a chance to try it out?

**Comment by sjquigley** [comment]
Hello Charly,

I have been away from my university.  I will be on Phoenix Code within the next few days. I’m excited. Thank you for your hard work.




Kind regards,

Stephen J. Quigley, Ph.D.
John C. Mascaro Faculty Lecturer in Sustainability
Associate Teaching Professor
University of Pittsburgh
English Department
Cathedral of Learning 628K
http://www.pitt.edu/~sjq4
***@***.******@***.***>

Pitt Senate Computing and
Information Technology
Committee Member

Have you heard about Pitt’s
new Digital Narrative and
Interactive Design Major?
https://www.dnid.pitt.edu































On Aug 14, 2024, at 3:44 PM, charly p abraham ***@***.***> wrote:



Hello @sjquigley<https://github.com/sjquigley>

Have you got a chance to try it out?
